rest.getinsomnia.Insomnia.json

```diff
@@ -0,0 +1,68 @@
+{
+    "app-id": "rest.getinsomnia.Insomnia",
+    "base": "io.atom.electron.BaseApp",
+    "base-version": "stable",
+    "runtime": "org.freedesktop.Platform",
+    "runtime-version": "1.6",
+    "sdk": "org.freedesktop.Sdk",
+    "command": "Insomnia",
+    "finish-args": [
+        "--socket=x11",
+        "--share=ipc",
+        "--socket=wayland",
+        "--device=dri",
+        "--share=network",
+        "--filesystem=home"
+    ],
+    "separate-locales": false,
+    "modules": [
+        {
+            "name": "Insomnia",
+            "buildsystem": "simple",
+            "build-commands": [
+                "install -Dm755 apply_extra /app/bin/apply_extra",
+                "install -Dm644 rest.getinsomnia.Insomnia.png /app/share/icons/hicolor/128x128/apps/rest.getinsomnia.Insomnia.png",
+                "install -Dm644 -t /app/share/applications rest.getinsomnia.Insomnia.desktop",
+                "install -Dm644 -t /app/share/metainfo rest.getinsomnia.Insomnia.appdata.xml",
+                "install -Dm755 insomnia.sh /app/bin/Insomnia"
+            ],
+            "sources": [
+                {
+                    "type": "script",
+                    "dest-filename": "apply_extra",
+                    "commands": [
+                        "tar xf insomnia.tar.gz",
+                        "rm -f insomnia.tar.gz",
+                        "mkdir -p /app/extra/Insomnia",
+                        "mv insomnia-6.0.1/* /app/extra/Insomnia"
+                    ]
+                },
+                {          
+                    "type": "extra-data",
+                    "filename": "insomnia.tar.gz",
+                    "only-arches": ["x86_64"],
+                    "url": "https://github.com/getinsomnia/insomnia/releases/download/v6.0.1/insomnia-6.0.1.tar.gz",
+                    "sha256": "b785ff6bd121fd01c5c2351c57f9024a905e9f5bb3f5705fd959c761e178b92c",
+                    "size": 60918635
+                },
+                {
+                    "type": "script",
+                    "dest-filename": "insomnia.sh",
+                    "commands": ["exec /app/extra/Insomnia/insomnia \"$@\""]
+                },
+                {
+                    "type": "file",
+                    "path": "rest.getinsomnia.Insomnia.desktop"
+                },
+                {
+                    "type": "file",
+                    "path": "rest.getinsomnia.Insomnia.appdata.xml"
+                },
+                {
+                    "type": "file",
+                    "path": "rest.getinsomnia.Insomnia.png"
```

Comment:
It's not present in upstream archive?

Reply:
Unfortunately *extra-data* archive don't provide icon, so I choose to put it here.
This icon is also available on app repository: https://github.com/getinsomnia/insomnia/tree/develop/packages/insomnia-app/app/icons/icons

It's good practice to download it from this source and not provide copy of it on Flathub?

Reply:
No, I don't think it matters. Just wanted to make sure.